pkg/resource/deploy/deployment_executor.go

```diff
@@ -462,15 +489,10 @@ func (ex *deploymentExecutor) handleSingleEvent(event SourceEvent) error {
 		for _, errored := range ex.stepExec.GetErroredSteps() {
 			ex.skipped.Add(errored.Res().URN)
 		}
-	outer:
 		for _, step := range steps {
-			for _, dep := range step.Res().Dependencies {
-				if ex.skipped.Contains(dep) {
-					step.Skip()
-					ex.skipped.Add(step.Res().URN)
```

Comment:
I think I would have wrote skipStepIfDependenciesAreSkipped as a pure query function (doesStepDependOn) then kept the `step.Skip(); ex.skipped.Add(step.Res().URN)` in this method.

Reply:
Good point, that is much nicer! :heavy_check_mark: